src/common/helpers/snap-builds.js

```diff
@@ -25,43 +25,116 @@
 // };
 
 export const BuildStatusColours = {
+  BLUE: 'blue',
   GREEN: 'green',
   YELLOW: 'yellow',
   RED: 'red',
   GREY: 'grey'
 };
 
+const BuildAndPubState = {
+  'NEVER_BUILT':     createState('Never built', BuildStatusColours.GREY, false, 8),
+  'BUILDING_SOON':   createState('Building soon', BuildStatusColours.GREY, 'ellipsis', 7),
+  'IN_PROGRESS':     createState('In progress', BuildStatusColours.BLUE, 'spinner', 2),
+  'FAILED_TO_BUILD': createState('Failed to build', BuildStatusColours.RED, 'cross', 1),
+  'WONT_PUBLISH':    createState('Built, won\'t be published', BuildStatusColours.GREEN, 'tick_outlined', 6),
+  'PUBLISH_SOON':    createState('Built, publishing soon', BuildStatusColours.GREY, 'tick', 2),
+  'PUBLISH_NOW':     createState('Built, publishing now', BuildStatusColours.BLUE, 'tick', 3),
```

Comment:
This is either a design bug or a Launchpad bug depending on which way you look at it, but as far as I can tell there's no way for BSI to detect any state that we might describe as "publishing now": it's either pending ("publishing soon") or it's failed or it's done, and LP doesn't really expose the in-progress states between those.  We may or may not regard this as a problem, and at any rate I guess it doesn't need to be dealt with in this PR.